src/NewPost.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+
+export class NewPost extends Component {
+  state = {
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: []
+  };
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+  };
+
+  render() {
+    return (
+      <form>
+        <input
+          type="text"
+          name="title"
+          placeholder="blog title"
+          value={this.state.title}
+          onChange={e => this.handleChange(e)}
```

Comment:
all these `onChange`s would work with just `onChange={this.handleChange}`. they should automatically get the `event` object, so there's no need to manually pass it